### DISCUSSION TITLE
Gradient surgery changes spoiler area?

### DISCUSSION THREAD

**Discussion by h3lg3** [imr-framework/pypulseq#215] [Q&A]
Hi, I have an issue understanding gradient surgery. 
Is it a problem that gradient surgery effects for example the area for spoiler gradients? It may also affect the slew rate of the gradients. 
Or am I misunderstanding something? 
I stumbled upon this when I combined the readout gradient with two spoiler gradients around it. I wonder in which cases this might cause issues and how to deal with that. I guess for spoiler gradients it is not a big deal if the area is changed slightly. 

Any insight would be appreciated. 

Code example: 
```
    # combine parts of the read gradient
    gc_times = np.array(
        [
            0,
            grad_ro_spr.rise_time,
            grad_ro_spr.flat_time,
            grad_ro_spr.fall_time,
            grad_ro.flat_time,
            grad_ro_spr.fall_time,
            grad_ro_spr.flat_time,
            grad_ro_spr.rise_time ] )
    gc_times = np.cumsum(gc_times)

    gr_amp = np.array([0, 
                       grad_ro_spr.amplitude, 
                       grad_ro_spr.amplitude,
                       grad_ro.amplitude,
                       grad_ro.amplitude,
                       grad_ro_spr.amplitude,
                       grad_ro_spr.amplitude,
                       0])
    gr = pp.make_extended_trapezoid(
        channel=channels.ro, 
        times=gc_times, 
        amplitudes=gr_amp)
```
**IDEA simulation**
![image](https://github.com/user-attachments/assets/8ecd021f-d09b-4acc-a7b3-feba69f7958d)

**Illustration**
![image](https://github.com/user-attachments/assets/c610206b-4b7e-4c23-bdc0-887c18c7e5aa)

**Comment by btasdelen** [comment]
I am confused about your expectation. `make_extended_trapezoid` takes the time points, and the amplitudes at those time points, as you provided. And looking at the code, you provided the amplitudes to generate the second gradient in your illustration. There is no `area` that needed to be kept fixed here. If you wanted to keep the area the same as the first illustration, you just need to either adjust the amplitudes you provide or timing to do so.

**Reply by h3lg3** [reply]
You are right, given my code, the second gradient is what you should expect. 
The reason for my question is more general about how to use "gradient surgery" properly and what you need to be careful about if you do so. 
The given example is a specific case, that I have seen sometimes in example code. In a first step, people define spoiler and readout gradients seperately and afterwards combine them using `make_extended_trapezoid`. I was wondering, if you do so, that you are at risk to change the spoiler area as I illustrated above. 
Is it a good idea to use "gradient surgery" to combine spoiler and readout gradients or should I not do so? Maybe it is better to use `add_gradients` not to change the total area? 
Thanks for your feedback. Below is an extended version of the sample code where I have seen gradient surgery. 

Kind regards
Helge

```
# STEP 1: Define readout gradient and spoiler 

grad_ro = pp.make_trapezoid(
    channel=channels.ro,
    system=system,
    amplitude=ro_amplitude,
    flat_time=ro_flattop_time, # Add gradient correction time and ADC correction time
    delay=t_sp
)

# Readout spoiler gradient
grad_ro_spr = pp.make_trapezoid(
    channel=channels.ro,
    system=system,
    area=grad_ro.area,  # grad_ro_spr.area = 0 why not same as set 0 here
    duration=t_sp,
    )


# STEP 2: Gradient surgery: combine parts of the read gradient

gc_times = np.array(
    [
        0,
        grad_ro_spr.rise_time,
        grad_ro_spr.flat_time,
        grad_ro_spr.fall_time,
        grad_ro.flat_time,
        grad_ro_spr.fall_time,
        grad_ro_spr.flat_time,
        grad_ro_spr.rise_time ] )
gc_times = np.cumsum(gc_times)

gr_amp = np.array([0, 
                   grad_ro_spr.amplitude, 
                   grad_ro_spr.amplitude,
                   grad_ro.amplitude,
                   grad_ro.amplitude,
                   grad_ro_spr.amplitude,
                   grad_ro_spr.amplitude,
                   0])
gr = pp.make_extended_trapezoid(
    channel=channels.ro, 
    times=gc_times, 
    amplitudes=gr_amp)
```
